src/Form/form-autosuggest.mdx

```diff
@@ -19,52 +19,77 @@ Form auto-suggest enables users to manually select or type to find matching opti
-      errorMessageText="Error, no selected value"
-      helpMessage="Select language"
-      value={selected}
-      onSelected={(value) => setSelected(value)}
-    >
-      <Form.AutosuggestOption>PHP</Form.AutosuggestOption>
-      <Form.AutosuggestOption>Java</Form.AutosuggestOption>
-      <Form.AutosuggestOption>Turbo Pascal</Form.AutosuggestOption>
-      <Form.AutosuggestOption>Flask</Form.AutosuggestOption>
-    </Form.Autosuggest>
+    <>
+      <Form.Group>
+        <Form.Label>
+          <h4>Programming language</h4>
+        </Form.Label>
+        <Form.Autosuggest
+          ref={autosuggestRef}
+          aria-label="form autosuggest"
+          helpMessage="Select language"
+          value={value}
+          onChange={(v) => setValue(v)}
+          valueRequired={valueRequired}
+          valueRequiredErrorMessageText="Error: value required"
+          selectionRequired={selectionRequired}
+          selectionRequiredErrorMessageText="Error: selection required"
+          customError={customError()}
+          customErrorMessageText="Error: selected language less than 50 years old"
+        >
+          <Form.AutosuggestOption id="javascript-option-id">JavaScript</Form.AutosuggestOption>
+          <Form.AutosuggestOption id="python-option-id">Python</Form.AutosuggestOption>
+          <Form.AutosuggestOption id="ruby-option-id">Ruby</Form.AutosuggestOption>
+          <Form.AutosuggestOption id="c-option-id">C</Form.AutosuggestOption>
+        </Form.Autosuggest>
+      </Form.Group>
+      <Form.Group>
+        <Form.CheckboxSet isInline>
+          <Form.Checkbox checked={valueRequired} onChange={e => setValueRequired(e.target.checked)}>Value Required</Form.Checkbox>
+          <Form.Checkbox checked={selectionRequired} onChange={e => setSelectionRequired(e.target.checked)}>Selection Required</Form.Checkbox>
+          <Form.Checkbox checked={customValidation} onChange={e => setCustomValidation(e.target.checked)}>Custom Validation</Form.Checkbox>
+        </Form.CheckboxSet>
+      </Form.Group>
+      <Collapsible styling="basic" title="Value details">
+        <Row>
+          <Col xs={3}><pre>userProvidedText:</pre></Col>
+          <Col><pre>{value.userProvidedText}</pre></Col>
+        </Row>
+        <Row>
+          <Col xs={3}><pre>selectionValue:</pre></Col>
+          <Col><pre>{value.selectionValue}</pre></Col>
+        </Row>
+        <Row>
+          <Col xs={3}><pre>selectionId:</pre></Col>
+          <Col><pre>{value.selectionId}</pre></Col>
+        </Row>
+      </Collapsible>
+      <Collapsible styling="basic" title="External modification">
+        User provided text
+        <Form.Control
+          onChange={(e) => setValue({
+            userProvidedText: e.target.value,
+            selectionValue: '',
+            selectionId: '',
+          })}
+          value={value.userProvidedText}
+        />
+        <Button onClick={forceUpdateErrorState}>Trigger validation</Button>
```

Comment:
nit: consider using `Stack` or CSS utility class to give some vertical whitespace between the `Form.Control` and the `Button`.

<img width="911" alt="image" src="https://github.com/openedx/paragon/assets/2828721/3d057f9b-dabe-44a4-83b3-846eefc4866e">

Reply:
addressed here https://github.com/openedx/paragon/compare/f8a4e5ee118199936c3ff6653e3dfb389d12fffa..9310aa04d988eba1ee524cf920e5ad54dd39590f#diff-3ce694685ee1cc2b274e86b07ff39240bff96ddf3862d4c5366d0f24ad863d38R81-R91